Scripts/RemediationScripts/Remediate-SetServiceBusMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,779 @@
+        Write-Host "Account Type: [$($context.Account.Type)]"
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    Write-Host "To set secure TLS version for Azure Service Bus(s) in the Subscription, Contributor on Subscription/RG or Azure Data Owner role on Service Bus Namespace(s) is required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::SingleDashLine)  
+
+    # list to store Service Bus Namespace details.
+    $serviceBusNamespacesDetails = @()
+
+    # To keep track of remediated and skipped resources	
+    $logRemediatedResources = @()	
+    $logSkippedResources = @()	
+
+    $controlId = "Azure_ServiceBus_DP_Use_Secure_TLS_Version"
+
+    if ($AutoRemediation) {
+        if (-not (Test-Path -Path $Path)) {	
+            Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)	
+            Write-Host $([Constants]::DoubleDashLine)	
+            return	
+        }
+
+        Write-Host "Fetching all Azure Service Bus Namespace(s) details failing for the [$($controlId)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+
+        # list to store Service Bus Namespace details.
+        $serviceBusNamespacesDetails = @()
+        
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlId -eq $_.ControlId };
+
+        $validServiceBusNampespaces = $resourceDetails.FailedResourceList | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }	
+        if (($resourceDetails | Measure-Object).Count -eq 0 -or ($validServiceBusNampespaces | Measure-Object).Count -eq 0) {
+
+            Write-Host "No Azure Service Bus Namespace(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)	
+            Write-Host $([Constants]::DoubleDashLine)	
+            return	
+        }
+        $validServiceBusNampespaces | ForEach-Object { 	
+            try {
+                $currentServiceBusNamespaceDetails = Get-AzServiceBusNamespace -ResourceGroupName $_.ResourceGroupName -NamespaceName $_.ResourceName -ErrorAction SilentlyContinue
+                $currentTLSVersion = ($currentServiceBusNamespaceDetails).MinimumTlsVersion
+                $serviceBusNamespacesDetails += $currentServiceBusNamespaceDetails  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
+                @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
+                @{N = 'ResourceName'; E = { $_.Name } }, 
+                @{N = 'MinimumTLSVersion'; E = { $currentTLSVersion } }
+            }
+            catch {
+                Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Warning)	
```

Comment:
this statement should be of error type imo

Reply:
Changed.